### PR TITLE
ProgressTableDetailView - progress table refactor

### PR DESCRIPTION
**Pull request by maureensturgeon**
Dependent on https://github.com/code-dot-org/code-dot-org/pull/38909

This is a new component which will replace the [DetailView](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/sectionProgress/detail/DetailView.jsx) a combined effort between @cforkish and @maureensturgeon, and was extracted from the full refactor work in #38309.

Unit tests were PR'd previously: https://github.com/code-dot-org/code-dot-org/pull/38642

ProgressTableDetailView:
Screen Shot 2021-02-04 at 12 16 01 PM

You can refer to [this tech spec](https://docs.google.com/document/d/1SS1DYiLObiccRrVTo3LuLSe5OFiWla2qhsIWlfMvPIo/edit#) to see how this component fits into the full table architecture.

## Links

- [spec]()
- [jira](https://codedotorg.atlassian.net/browse/LP-1682)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgress/progressTables/ProgressTableDetailView.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  };
+
+  constructor(props) {
+    super(props);
+    this.levelIconFormatter = this.levelIconFormatter.bind(this);
+    this.detailCellFormatter = this.detailCellFormatter.bind(this);
+  }
+
+  getTableWidth(lessons) {
+    return lessons.reduce((lessonSum, lesson) => {
```
Comment (jamescodeorg) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableDetailView.jsx]: Maybe rename `lessonSum` to `width` or `totalWidth`?

File: apps/src/templates/sectionProgress/progressTables/ProgressTableDetailView.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  getCurrentScriptData,
+  setLessonOfInterest
+} from '@cdo/apps/templates/sectionProgress/sectionProgressRedux';
+import {sectionDataPropType} from '@cdo/apps/redux/sectionDataRedux';
+import ProgressTableContainer from './ProgressTableContainer';
+import ProgressTableDetailCell from './ProgressTableDetailCell';
+import ProgressTableLevelIcon from './ProgressTableLevelIcon';
+import ProgressLegend from '@cdo/apps/templates/progress/ProgressLegend';
+
+class ProgressTableDetailView extends React.Component {
```
Comment (jamescodeorg) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableDetailView.jsx:21]: (minor) A header comment here would be useful.  Things that might be useful to touch upon:
- This component displays the progress bubbles for all levels in all lessons in a single script.
- It glues together the components specific to detailed view (ProgressTableLevelIcon, ProgressTableDetailCell) with the shared progress view components (ProgressTableContainer)
- There is also a sister component ProgressTableSummaryView that presents a more compact view of the same data.

File: apps/src/templates/sectionProgress/progressTables/ProgressTableDetailView.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    this.detailCellFormatter = this.detailCellFormatter.bind(this);
+  }
+
+  getTableWidth(lessons) {
+    return lessons.reduce((lessonSum, lesson) => {
+      return lessonSum + ProgressTableDetailCell.widthForLevels(lesson.levels);
+    }, 0);
+  }
+
+  levelIconFormatter(_, {columnIndex}) {
```
Comment (jamescodeorg) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableDetailView.jsx]: (minor) Might be helpful to indicate that this is for the icons in the second header row.  Or include the word 'header' in the name?

**Review by jamescodeorg — APPROVED** [review]
A few cosmetic suggestions, but lgtm.

File: apps/src/templates/sectionProgress/progressTables/ProgressTableDetailView.jsx (hunk 2 above)
Comment (maureensturgeon) [reply apps/src/templates/sectionProgress/progressTables/ProgressTableDetailView.jsx:21]: I should have seen this coming! Will update.

**Review by cforkish — APPROVED** [review]

**Pull request merged**